### DISCUSSION TITLE
Automating names to faces through file names?

### DISCUSSION THREAD

**Discussion by tomhubeek** [immich-app/immich#6938] [Q&A]
Hello!

We've just started out with immich and I've got a whole heap of high quality portrait style photos I'd want to use to train the our instance on. At the moment I believe we need to upload the portrait, wait for the face to be found and then name it with the correct name.

Is there a way we can name the portrait image as "John Smith.jpg" and have that face set as "John Smith" without having to type each one manually? Any ideas on speeding up this process would be appreciated!

Cheers!

**Comment by aisbergde** [comment]
it could be complicated to extract person names from file names (a specific syntax would be required, conflict handling and so on).
I could imagine to use external software to extract the filename and to add this as metadata. And then this metadata could be used by immich. I have seen a feature request. But I am not sure, if some developer will ever implement this.

So, for now the easiest way will be to let immich detect and recognize faces, then to open the face group, select the first asset, show details of the asset and copy and paste the filename from this asset.
